packages/db/index.js

```diff
@@ -763,56 +763,53 @@ export class DBClient {
   /**
    * Creates a Pin Request.
    *
-   * @param {import('./db-client-types').PAPinRequestUpsertInput} pinRequest
-   * @return {Promise<import('./db-client-types').PAPinRequestUpsertOutput>}
+   * @param {import('./db-client-types').PsaPinRequestUpsertInput} pinRequestData
+   * @return {Promise<import('./db-client-types').PsaPinRequestUpsertOutput>}
    */
-  async createPAPinRequest (pinRequest) {
-    /** @type { import('./db-client-types').PAPinRequestUpsertOutput } */
-
-    // TODO is there a better way to avoid 2 queries?
-    // ie. before insert trigger https://dba.stackexchange.com/questions/27178/handling-error-of-foreign-key
-    /** @type {{data: {cid: string}}} */
-    const { data: content } = await this._client
-      .from('content')
-      .select('cid')
-      .eq('cid', pinRequest.cid)
-      .single()
-
-    const toInsert = {
-      requested_cid: pinRequest.requestedCid,
-      auth_key_id: pinRequest.authKey,
-      name: pinRequest.name
-    }
-
-    // If content already exists updated foreigh key
-    if (content?.cid) {
-      toInsert.content_cid = content.cid
-    }
+  async createPsaPinRequest (pinRequestData) {
+    const now = new Date().toISOString()
 
-    /** @type {{data: import('./db-client-types').PAPinRequestItem, error: PostgrestError }} */
-    const { data, error } = await this._client
-      .from(PAPinRequestTableName)
-      .insert(toInsert)
-      .select(pinRequestSelect)
-      .single()
+    /** @type {{ data: string, error: PostgrestError }} */
+    const { data: pinRequestId, error } = await this._client.rpc('create_psa_pin_request', {
+      data: {
+        auth_key_id: pinRequestData.authKey,
+        content_cid: pinRequestData.contentCid,
+        source_cid: pinRequestData.sourceCid,
+        name: pinRequestData.name,
+        dag_size: pinRequestData.dagSize,
+        inserted_at: pinRequestData.created || now,
+        updated_at: pinRequestData.updated || now,
```

Comment:
Do we need to generate and send these two fields? The DB has defaults.

Reply:
Yeah, I agree, I was "annoyed" by this as well.

IIRC we decided to keep it consistent with the `createUpload` behavior.
We're calling a remote procedure here and we were not quite sure what the behavior would be with defaults (when it comes to PostgreSQL) so we assumed there was some issue with that given your `createUpload` implementation.